modules/test/protocol/python/src/protocol_modbus.py

```diff
@@ -267,7 +267,7 @@ def validate_device(self):
       compliant = (holding_reg is not None or input_reg is not None
                    or coils is not None or discrete_inputs is not None)
     else:
-      compliant = False
+      compliant = 'Feature Not Present'

```

Comment:
Replacing the failure with a feature not present is not correct. There is no way to determine if a device supports Modbus feature if not discovered, this needs further configuration.